versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/TxDatabaseAdapter.java

```diff
@@ -1918,4 +1868,26 @@ public void deleteAttachments(Stream<ContentAttachmentKey> keys) {
       throw new RuntimeException(e);
     }
   }
+
+  @FunctionalInterface
+  interface ThrowingStreamResult<R, E extends Exception> {
+    @MustBeClosed
+    Stream<R> process(ConnectionWrapper conn) throws E;
+  }
+
+  private <R, E extends Exception> Stream<R> withConnectionWrapper(ThrowingStreamResult<R, E> x)
+      throws E {
+    ConnectionWrapper conn = borrowConnection();
+    boolean failed = true;
+    try {
+      @SuppressWarnings("MustBeClosedChecker")
```

Comment:
am i missing something or why are we suppressing this check here when we intentionally added the `@MustBeClosed` above? this is the only place where `process` is getting called, right?

did we forget to annotate this helper-method with `@MustBeClosed` itself?

Reply:
yup